### PR TITLE
chore(medusa): Migrate price list repository

### PR DESCRIPTION
**Pull request by adrien2p**
FIXES CORE-1309

### CODE REVIEW

File: packages/medusa/src/repositories/price-list.ts [hunk 1]
```diff
@@ -1,39 +1,60 @@
...
+          name: ILike(`%${q}%`),
+        },
+        {
+          ...query_.where,
+          description: ILike(`%${q}%`),
+        },
+        {
+          ...query_.where,
+          customer_groups: {
+            ...query_.where.customer_groups,
```
Comment (riqwan) [inline packages/medusa/src/repositories/price-list.ts:50]: Q: In line 18, you're deleting customer_groups, why isn't this failing for cases where `groups is falsey`?

**Review by riqwan — APPROVED** [review]
Added one question, looks good otherwise

File: packages/medusa/src/repositories/price-list.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/repositories/price-list.ts:50]: just lucky I suppose 😂 , I can see some other thing to fix as well

File: packages/medusa/src/repositories/price-list.ts [hunk 2]
```diff
@@ -1,39 +1,69 @@
...
+        query_.where.push({
+          ...query_.where,
+          customer_groups: {
+            id: In(groups.value),
+            name: ILike(`%${q}%`),
+          },
+        })
+      } else {
+        query_.relations = query_.relations ?? {}
+        query_.relations.customer_groups =
```
Comment (riqwan) [inline packages/medusa/src/repositories/price-list.ts]: q: why would customer_groups be true for all cases except search? We're doing this above in line 22 as well

File: packages/medusa/src/repositories/price-list.ts (hunk 1 above)
Comment (riqwan) [reply packages/medusa/src/repositories/price-list.ts]: Much better. This opens up a new question. :D

File: packages/medusa/src/repositories/price-list.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/repositories/price-list.ts]: It is outdated ^^

Comment (riqwan) [reply packages/medusa/src/repositories/price-list.ts]: ahah, done done.

**Review by riqwan — APPROVED** [review]

File: packages/medusa/src/repositories/price-list.ts (hunk 2 above)
Comment (adrien2p) [reply packages/medusa/src/repositories/price-list.ts]: i ve clean it up a bit top avoid redundancy and incoherences, wdyt?

Comment (riqwan) [reply packages/medusa/src/repositories/price-list.ts]: 🤌

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**